### PR TITLE
option to show both elapsed and remaining time simultaneously

### PR DESCRIPTION
**Pull request by Be-ing**
Add a third option for WNumberPos to show both the elapsed and remaining time on a deck. When this is chosen, the format is "elapsed time / remaining time".

### CODE REVIEW

File: src/preferences/dialog/dlgprefcontrolsdlg.ui [hunk 1]
```diff
@@ -482,6 +482,16 @@
        </attribute>
       </widget>
      </item>
+    <item row="9" column="3">
```
Comment (esbrandt) [inline src/preferences/dialog/dlgprefcontrolsdlg.ui]: The introduction of a 3rd column throws off the positioning/aligning of the existing options. Instead, like with the tooltips options, use a horizontal layout for the time display options.

File: src/widget/wnumberpos.cpp [hunk 2]
```diff
@@ -45,73 +45,97 @@ WNumberPos::WNumberPos(const char* group, QWidget* parent)
...
+                        -dPosSecondsElapsed, mixxx::Duration::Precision::CENTISECONDS));
+        }
+    } else if (m_displayMode == DisplayMode::Remaining) {
+        setText(QLatin1String("-") % mixxx::Duration::formatSeconds(
+                    dPosSecondsRemaining, mixxx::Duration::Precision::CENTISECONDS));
+    } else if (m_displayMode == DisplayMode::ElapsedAndRemaining) {
+        if (dPosSecondsElapsed >= 0.0) {
+            setText(mixxx::Duration::formatSeconds(
+                        dPosSecondsElapsed, mixxx::Duration::Precision::CENTISECONDS)
+                    % QLatin1String(" / -") %
```
Comment (esbrandt) [inline src/widget/wnumberpos.cpp]: To me the ``/`` separator is unneeded.
Depending on the font, i found it harder to read with it, we already have the ``-``in front of the remaining time display as separator, so 3 spaces and ``-`` should do it.

**Comment by esbrandt** [comment]
Thanks for adding a long standing request.
Related to https://bugs.launchpad.net/mixxx/+bug/1081456

**Comment by esbrandt** [comment]
LGTM

File: src/preferences/dialog/dlgprefcontrols.cpp [hunk 3]
```diff
@@ -70,9 +70,14 @@ DlgPrefControls::DlgPrefControls(QWidget * parent, MixxxMainWindow * mixxx,
     if (!m_pConfig->exists(ConfigKey("[Controls]","PositionDisplay")))
         m_pConfig->set(ConfigKey("[Controls]","PositionDisplay"),ConfigValue(0));
 
-    if (m_pConfig->getValueString(ConfigKey("[Controls]", "PositionDisplay")).toInt() == 1) {
+    int positionDisplayType = m_pConfig->getValueString(
```
Comment (daschuer) [inline src/preferences/dialog/dlgprefcontrols.cpp]: You can combine this line and the line above by 
```
int positionDisplayType = m_pConfig->getValue(ConfigKey("[Controls]", "PositionDisplay"), 0);
```

File: src/preferences/dialog/dlgprefcontrols.cpp [hunk 4]
```diff
@@ -70,9 +70,14 @@ DlgPrefControls::DlgPrefControls(QWidget * parent, MixxxMainWindow * mixxx,
...
+            ConfigKey("[Controls]", "PositionDisplay")).toInt();
+    if (positionDisplayType == 1) {
         radioButtonRemaining->setChecked(true);
         m_pControlTrackTimeDisplay->set(1.0);
```
Comment (daschuer) [inline src/preferences/dialog/dlgprefcontrols.cpp]: Please use constexpr for these magic numbers.

Comment (Be-ing) [inline src/preferences/dialog/dlgprefcontrols.cpp]: I don't understand how a constexpr would help here. Do you mean an enum class?

Comment (daschuer) [inline src/preferences/dialog/dlgprefcontrols.cpp]: You cannot convert a enum class to a double without a cast. 
The idea was to define 
constexpr double k... = 1.0 
to get around this. 

Everything that replaces the 1.0 by a readable text works for me here, I do not care much about type safety here. 
You already use a private enum class inside the widget with the same values. Maybe you find common solution for both.

**Comment by MK-42** [comment]
Tested on Shade, Deere and LateNight, LGTM

**Comment by daschuer** [comment]
Clang build fails: 

```
In file included from src/preferences/dialog/dlgprefcontrols.cpp:27:
src/preferences/dialog/dlgprefcontrols.h:34:11: error: redefinition of 'Duration' as different kind of symbol
namespace Duration {
          ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/usr/include/MacTypes.h:286:41: note: previous definition is here
typedef SInt32                          Duration;
                                        ^
```

**Comment by Be-ing** [comment]
Build fixed on all platforms.

**Comment by Be-ing** [comment]
Ready for merge?

File: src/preferences/dialog/dlgprefcontrols.cpp [hunk 5]
```diff
@@ -70,12 +70,23 @@ DlgPrefControls::DlgPrefControls(QWidget * parent, MixxxMainWindow * mixxx,
     if (!m_pConfig->exists(ConfigKey("[Controls]","PositionDisplay")))
         m_pConfig->set(ConfigKey("[Controls]","PositionDisplay"),ConfigValue(0));
```
Comment (daschuer) [inline src/preferences/dialog/dlgprefcontrols.cpp]: Here we should also use the TrackTime::DisplayMode class.
This value does not match the value set in slotResetToDefaults()

Comment (Be-ing) [inline src/preferences/dialog/dlgprefcontrols.cpp]: This is a common issue whenever someone works on a preferences dialog. Perhaps as part of @Pegasus-RPG's refactoring of the preferences we can come up with a better solution.

**Comment by daschuer** [comment]
Thank you for working on that (and all the other stuff) 
LGTM

**Pull request merged**